source/lac/trilinos_sparse_matrix.cc

```diff
@@ -1102,7 +1099,6 @@ namespace TrilinosWrappers
   SparseMatrix::clear_rows (const std::vector<size_type> &rows,
                             const TrilinosScalar          new_diag_value)
   {
-    compress();
```

Comment:
This will change how this function works. A code that was valid before may now throw an exception.